src/language/from_relay/mod.rs

```diff
@@ -662,22 +674,59 @@ fn compile_expression(
         let arg = tuple_get_item
             .tuple
             .clone()
-            .downcast::<tvm::ir::relay::Call>()
-            .unwrap();
-        assert_eq!(
-            arg.op
-                .clone()
-                .upcast::<tvm::ir::expr::BaseExpr>()
-                .downcast::<tvm::ir::op::Op>()
-                .unwrap()
-                .name
-                .as_str(),
-            Ok("nn.batch_norm")
-        );
-        assert_eq!(tuple_get_item.index, 0);
+            .downcast::<tvm::ir::relay::Call>();
+
```

Comment:
There's a comment a few lines up ([line 670](https://github.com/gussmith23/glenside/pull/94/files#diff-662d20b9cf2210216fea2642a0736aa1edf391b04248d7b90210a09199df78ebR670)) that should be updated, because it's no longer true!

Recapping my own understanding: the tuple_get_item code looks this way because I used to only parse tuple_get_item in the case where it was a tuple_get_item getting the first item of the result of a `nn.batch_norm`, in which case I just compiled it straight to a `BatchNormInference`. This is a hack, plain and simple, but it's a useful hack, and is still a useful hack, so we need to keep it around. At the same time, we need to make the code more general, now that we actually want to compile tuples.

Here's what I think we need: we should keep code for the special case where we have a tuple_get_item accessing index 0 of a batch norm, and we should have that compile to a Glenside RelayOperator batch norm (as currently happens), though we should hide it behind the `simplify_batch_norm_for_inference` argument that gets passed in to from_relay. This feature flag is also checked in the code that compiles an `nn.batch_norm` expression. If the flag is set, that code will know to compile an `nn.batch_norm` to a single tensor output; if it's not set, we should panic (because we haven't yet implemented the full Relay batch norm). That's already done, though. It seems like you do all of this already, _except_ you need to add a check for `simplify_batch_norm_for_inference_hack`, maybe in the if condition on line 681?